### PR TITLE
DOC: Update the pandas.DatetimeIndex docstring

### PR DESCRIPTION
**Pull request by Hori75**
- [x] closes https://github.com/pandanistas/pandanistas_sprint_ui2020/issues/7
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Output of ```python scripts/validate_docstrings.py pandas.DatetimeIndex```
```
################################################################################
################################## Validation ##################################
################################################################################

1 Errors found:
	No examples section found
```

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -78,21 +78,26 @@ def _new_DatetimeIndex(cls, d):
...
     freq : str or pandas offset object, optional
         One of pandas date offset strings or corresponding objects. The string
         'infer' can be passed in order to set the frequency of the index as the
         inferred frequency upon creation.
     tz : pytz.timezone or dateutil.tz.tzfile
+        Timezone of the datetime.
+    normalize : bool, default False
+        Normalize start/end dates to midnight before generating date range.
+    closed : {None, 'left', 'right'}, optional
+        Make the interval closed with respect to the given frequency to
```
Comment (MomIsBestFriend) [inline pandas/core/indexes/datetimes.py]: I *think* that you can remove the ```None``` here, since we are saying that this parameter is optional.

**Review by WillAyd — APPROVED** [review]
lgtm @jbrockmendel care to take a look?

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -78,21 +78,26 @@ def _new_DatetimeIndex(cls, d):
 )
 class DatetimeIndex(DatetimeTimedeltaMixin):
     """
-    Immutable ndarray of datetime64 data, represented internally as int64, and
-    which can be boxed to Timestamp objects that are subclasses of datetime and
-    carry metadata such as frequency information.
+    Immutable ndarray of datetime64 data.
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: not actually an ndarray anymore. maybe "ndarray-like"?

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -78,21 +78,26 @@ def _new_DatetimeIndex(cls, d):
...
+
+    Represented internally as int64, and which can be boxed to Timestamp objects
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: its a stretch to say "represented internally as int64" is accurate.

mentioning that DTI has freq is good, but id rather avoid mentioning it for Timestamp.

File: pandas/core/indexes/datetimes.py [hunk 4]
```diff
@@ -78,21 +78,26 @@ def _new_DatetimeIndex(cls, d):
...
     data : array-like (1-dimensional), optional
         Optional datetime-like data to construct index with.
-    copy : bool
-        Make a copy of input ndarray.
     freq : str or pandas offset object, optional
         One of pandas date offset strings or corresponding objects. The string
         'infer' can be passed in order to set the frequency of the index as the
         inferred frequency upon creation.
     tz : pytz.timezone or dateutil.tz.tzfile
+        Timezone of the datetime.
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: we also support stdlib tzinfos, and you can pass strings that can be parsed to timezones

File: pandas/core/indexes/datetimes.py [hunk 5]
```diff
@@ -78,21 +78,26 @@ def _new_DatetimeIndex(cls, d):
...
+    normalize : bool, default False
+        Normalize start/end dates to midnight before generating date range.
+    closed : {'left', 'right'}, optional
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: should "both:" be in here?

Comment (jbrockmendel) [reply pandas/core/indexes/datetimes.py]: can we avoid the word "interval" below?  dont want ambiguity with IntervalIndex

File: pandas/core/indexes/datetimes.py [hunk 6]
```diff
@@ -107,12 +112,16 @@ class DatetimeIndex(DatetimeTimedeltaMixin):
           times)
         - 'NaT' will return NaT where there are ambiguous times
         - 'raise' will raise an AmbiguousTimeError if there are ambiguous times.
-    name : object
-        Name to be stored in the index.
     dayfirst : bool, default False
         If True, parse dates in `data` with the day first order.
     yearfirst : bool, default False
         If True parse dates in `data` with the year first order.
+    dtype : numpy.dtype or DatetimeTZDtype
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: or str

File: pandas/core/indexes/datetimes.py [hunk 7]
```diff
@@ -107,12 +112,16 @@ class DatetimeIndex(DatetimeTimedeltaMixin):
...
+        Note that the only NumPy dtype allowed is ‘datetime64[ns]’.
+    copy : bool
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: default?

File: pandas/core/indexes/datetimes.py [hunk 8]
```diff
@@ -107,12 +112,16 @@ class DatetimeIndex(DatetimeTimedeltaMixin):
...
+        Make a copy of input ndarray.
+    name : object
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py]: Label

File: pandas/core/indexes/datetimes.py (hunk 5 above)
Comment (Hori75) [reply pandas/core/indexes/datetimes.py]: I think `None` means both.

**Comment by datapythonista** [comment]
Thanks for the improvements to this docstring @Hori75

**Pull request merged**